lib/zstd.h

```diff
@@ -392,28 +392,50 @@ ZSTDLIB_API size_t ZSTD_DStreamOutSize(void);   /*!< recommended size for output
 
 
 
-#if defined(ZSTD_STATIC_LINKING_ONLY) && !defined(ZSTD_H_ZSTD_STATIC_LINKING_ONLY)
-#define ZSTD_H_ZSTD_STATIC_LINKING_ONLY
-
 /****************************************************************************************
  *   ADVANCED AND EXPERIMENTAL FUNCTIONS
  ****************************************************************************************
- * The definitions in this section are considered experimental.
- * They should never be used with a dynamic library, as prototypes may change in the future.
+ * The definitions in the following section are considered experimental.
  * They are provided for advanced scenarios.
+ * They should never be used with a dynamic library, as prototypes may change in the future.
  * Use them only in association with static linking.
  * ***************************************************************************************/
 
+#if defined(ZSTD_STATIC_LINKING_ONLY) && !defined(ZSTD_H_ZSTD_STATIC_LINKING_ONLY)
+#define ZSTD_H_ZSTD_STATIC_LINKING_ONLY
+
+
+/****************************************************************************************
+ *   Candidate API for promotion to stable status
+ ****************************************************************************************
+ * The following symbols and constants are currently considered
+ * to join "stable API" status by v1.4.0.
+ * The intention is that they should be able to become stable "as is", with no further modification.
+ * It is also last chance to gather comments / suggestions on this API,
+ * as the API is locked once reaching "stable" status.
+ * ***************************************************************************************/
+
 ZSTDLIB_API int ZSTD_minCLevel(void);  /*!< minimum negative compression level allowed */
 
 /* ---  Constants  ---*/
-#define ZSTD_MAGICNUMBER            0xFD2FB528   /* v0.8+ */
-#define ZSTD_MAGIC_DICTIONARY       0xEC30A437   /* v0.7+ */
-#define ZSTD_MAGIC_SKIPPABLE_START  0x184D2A50U
+
+/* all magic numbers are supposed read/written to/from files/memory using little-endian convention */
+#define ZSTD_MAGICNUMBER            0xFD2FB528    /* valid since v0.8.0 */
+#define ZSTD_MAGIC_DICTIONARY       0xEC30A437    /* valid since v0.7.0 */
+#define ZSTD_MAGIC_SKIPPABLE_START  0x184D2A50    /* all 16 values, from 0x184D2A50 to 0x184D2A5F, signal the beginning of a skippable frame */
+#define ZSTD_MAGIC_SKIPPABLE_MASK   0xFFFFFFF0
+
+/* note : should this limit be smaller ? like 23 (8 MB) as suggested in the spec ?
+ * can it be different from zstd cli default limit, which is designed to match --ultra and --long default (27) ? */
+#define ZSTD_WINDOWLOG_LIMIT_DEFAULT 27   /* by default, the streaming decoder will refuse any frame
```

Comment:
`ZSTD_dParam_getBounds()` will only provide the limits of `ZSTD_d_windowLogMax`, not its default value.

Therefore, providing such a value dynamically from the library seems to require a dedicated prototype.
It somehow _feel_ overkill but I don't see an escape here. 
The value seems necessary, if only for documentation, so that users can understand how reacts the library on receiving frames with gigantic window sizes.

An alternative could be to create a "generic" get function,
such as `ZSTD_dParam_getDefault()`, 
valid for any `ZSTD_dParameter` (well, there are only 2 currently, and only `ZSTD_d_windowLogMax` matters).

However, note that the equivalent variant on the compression side, `ZSTD_cParam_getDefault()`, does not exist.
I suspect creating one variant on one side would warrant creating the other one, for consistency ?
Obviously, the compression variant is more complex, as it contains a lot more parameters.

And now, replacing this simple constant generates a surprisingly large amount of refactoring ...

Reply:
Whatever the solution (presence or not of a function to query this value),
the macro constant itself must go into _static_linking_only_ area.